src/OpenDiffix.Core/Planner.fs

```diff
@@ -149,4 +121,37 @@ let private filterJunk targetList plan =
 // ----------------------------------------------------------------
 
 let plan query =
-  query |> planQuery |> filterJunk query.TargetList
+  let selectedExpressions = query.TargetList |> List.map (fun column -> column.Expression)
+  let orderByExpressions = query.OrderBy |> List.map (fun (OrderBy(expression, _, _)) -> expression)
+  let expressions = query.Having :: selectedExpressions @ orderByExpressions
+  let aggregators = expressions |> collectAggregates |> List.distinct
+  let groupingLabels = query.GroupBy |> List.distinct
+
+  let expressionProjector, rowsAggregatorPlanner =
+    if groupingLabels.IsEmpty && aggregators.IsEmpty then
+      match query.AnonymizationContext with
+      | Some anonymizationContext when anonymizationContext.AnonymizationParams.UseAdaptiveBuckets ->
+        projectExpression selectedExpressions, planAdaptiveBuckets selectedExpressions anonymizationContext
```

Comment:
Do we verify what `selectedExpressions` can contain? Is it the same restrictions that apply to group labels?

What if we have an expr such as `2 * floor_by(age, 10)`? A smart planner might pull the `floor_by` into AB plan and `2 *` to a parent projection. This is rather complex so we might want to rely on clients doing further projections in an upper query.

Reply:
Restrictions are validated by the analyzer (and yes, they are the same as for the grouping expressions).
An expression like `2 * floor_by(age, 10)` wouldn't be allowed. Or are you referring to something else?